07-errors.md

```diff
@@ -0,0 +1,193 @@
+        - `ErrIllegalArgument` if the block isn't in memory, etc.
+    - `read`
+        - `ErrInvalidHandle` if the handle isn't known.
+        - `ErrIllegalArgument` if the passed buffer isn't valid, in memory, etc.
+    - `stat`
+        - `ErrInvalidHandle` if the handle isn't known.
+    - `cid`
+        - `ErrInvalidHandle` if the handle isn't known.
+        - `ErrIllegalCid` hash code and/or hash length aren't allowed.
+            - TODO: just use illegal argument?
+        - `ErrIllegalArgument` if the passed buffer isn't valid, in memory, etc.
+- `validation*` moved into the actor runtime
+- `self`
+    - `root`
+        - `ErrIllegalActor` actor hasn't set the root yet, or has been deleted.
+        - `ErrIllegalArgument` if the passed buffer isn't valid, in memory, etc.
+    - `set_root`
+        - `ErrIllegalActor` actor has been deleted
+        - `ErrInvalidHandle` specified root CID is not in the reachable set.
+    - `self_destruct`
+        - `ErrNotFound` beneficiary isn't found
+        - `ErrIllegalActor` beneficiary is self
+        - `ErrIllegalArgument` if the passed address buffer isn't valid, in memory, etc.
+- `message*` cannot fail
+- `network*` cannot fail
+- `actor`
+    - `resolve_address`
+        - `ErrNotFound` target actor doesn't exist
+        - `ErrIllegalArgument` if the passed address buffer isn't valid, in memory, etc.
+    - `get_actor_code_cid`
+        - `ErrNotFound` target actor doesn't exist
+        - `ErrIllegalArgument` if the passed address buffer isn't valid, in memory, etc.
+    - `new_actor_address` TODO (likely needs to go)
+    - `create_actor` TODO
+- `crypto`
+    - `verify_signature`
+        - `ErrIllegalArgument` signature, address, or plaintext buffers are invalid.
+    - `hash_blake2b` (consider removing after benchmarking WASM)
+        - `ErrIllegalArgument` invalid buffer
+    - `verify_seal`
+        - `ErrIllegalArgument` anything
+    - `verify_post`
+        - `ErrIllegalArgument` anything
+    - `compute_unsealed_sector_cid`
+        - TODO: new error for unknown proof type?
+        - `ErrIllegalArgument` everything else
+    - `verify_consensus_fault`
+        - `ErrLimit` exceeded lookback limit finding block.
+            - TODO is this useful?
+        - `ErrIllegalArgument` something else
+    - `verify_aggregate_seals`
+        - `ErrLimit` exceeds seal aggregation limit.
+        - `ErrIllegalArgument` something is malformed
+    - `batch_verify_seals`
+        - `ErrIllegalArgument` if malformed params
+- `rand`
+    - `get_chain_randomness`, `get_beacon_randomness`
+        - `ErrLimit` lookback exceeds limit.
+        - `ErrIllegalArgument` invalid buffer, etc.
+- `gas`
+    - `charge_gas`
+        - `ErrIllegalArgument` invalid name buffer.
+- `send`
+    - `send`
+        - `ErrNotFound` target actor does not exist and cannot be created.
+            - The `Executor` turns this into a `SysErrInvalidReceiver` exit code.
```

Comment:
Can you expand on what this actually means? The `Executor` is the "tip" of the execution structure, so this only applies to the chain message, but not to syscalls, so this reference here is a bit bizarre?